File: cpp/leetcode/409.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        unordered_map<char, int> words;
        for(auto ch : s) {
            words[ch]++;
        }
        bool flag = false;
        int res = 0;
        for(auto it : words){
            if(it.second % 2 == 0) {
                res += it.second;
            } else {
                res += flag ? it.second - 1 : it.second;
                flag = true;
            }
        }
        return res;
    }
};

File: python/leetcode/409.py
class Solution(object):
    def longestPalindrome(self, s):
        from collections import Counter
        t = Counter(s)
        flag = False
        res = 0
        for key, val in t.iteritems():
            if val % 2 == 1:
                flag = True
            res += val / 2 * 2
        if flag:
            res += 1
        return res
